README.md

```diff
@@ -52,70 +92,117 @@ cloudscraper.get('http://website.com/', function(error, response, body) {
+};
+
+cloudscraper(options, function(err, response, body) {
+  console.log(response)
 });
 ```
 
-## Error object
-Error object has following structure:
-```
-var error = {errorType: 0, error:...};
+## Advanced usage
+Cloudscraper wraps request and request-promise, so using cloudscraper is pretty much like using those two libraries.
+ - Cloudscraper exposes [the same request methods as request](https://github.com/request/request#requestmethod):
+ `cloudscraper.get(options, callback)`
+ `cloudscraper.post(options, callback)`
+ `cloudscraper(uri)`
+ Please refer to request's documentation for further instructions
+ - Cloudscraper uses request-promise, promise chaining is done exactly the same as described in [docs](https://github.com/request/request-promise#cheat-sheet):
+ ```
+  cloudscraper(options)
+    .then(function (htmlString) {
+    })
+    .catch(function (err) {
+    });
+  ```
+
+## Default options
+Cloudscraper exposes following options that areq required by default but might be changed. Please note that default options increase chances of correct work.
+
 ```
+var options = {
+  uri: 'https://website',
+  jar: requestModule.jar(), // Custom cookie jar
+  headers: {
+    // User agent, Cache Control and Accept headers are required
+    'User-Agent': 'Ubuntu Chromium/34.0.1847.116 Chrome/34.0.1847.116 Safari/537.36',
+    'Cache-Control': 'private',
+    'Accept': 'application/xml,application/xhtml+xml,text/html;q=0.9, text/plain;q=0.8,image/png,*/*;q=0.5'
+  },
+  // Cloudflare requires a delay of 5 seconds, so wait for at least 6.
+  cloudflareTimeout: 6000,
+  // followAllRedirects - follow non-GET HTTP 3xx responses as redirects
+  followAllRedirects: true,
+  // Support only this max challenges in row. If CF returns more, throw an error
+  challengesToSolve: 3
+};
+
+cloudscraper(options, function(error, response, body) {
+  console.log(body)
+});
 
+```
+## Error object
+Cliudscraper error object inherits from `Error` has following fields:
+  * `name` - `RequestError`/`CaptchaError`/`CloudflareError`/`ParserError`
+  * `options` - The request options
+  * `cause` - An alias for `error`
+  * `response` - The request response
+  * `errorType` - Custom error code
 Where `errorType` can be following:
  - `0` if request to page failed due to some native reason as bad url, http connection or so. `error` in this case will be error [event](http://nodejs.org/api/http.html#http_class_http_server)
  - `1` cloudflare returned captcha. Nothing to do here. Bad luck
  - `2` cloudflare returned page with some inner error. `error` will be `Number` within this range `1012, 1011, 1002, 1000, 1004, 1010, 1006, 1007, 1008`. See more [here](https://support.cloudflare.com/hc/en-us/sections/200038216-CloudFlare-Error-Messages)
  - `3` this error is returned when library failed to parse and solve js challenge. `error` will be `String` with some details. :warning: :warning: __Most likely it means that cloudflare have changed their js challenge.__
```

Comment:
If the vm throws an error, `error.cause` will be an `error` not a string. I think this should say something like: Do not always rely on `error.cause` to be an error, it can be a string. We could guarantee that `error.error` will always error by either constructing a new error or by the following: `error.error = error` but that seems odd, doesn't it? Hmm... Maybe we should also make `error.cause` always a string.

Reply:
Frankly speaking sounds like we are over complicating things a bit :) Let's leave as it is and then if someone complains, it can get fixed in one way or another